### ISSUE TITLE
Visual jitter?

### ISSUE THREAD

**Issue by varmanishant** [vifm/vifm.vim#20]
When I switch to `vifm`, I can see the terminal emulator -- especially the command `vim` - which was used to open `vim` -- for a split second (almost non-noticeable).

Is there a way to prevent this visual jitter? `dirvish` for example doesn't do this. I don't know if it is because of `:terminal`'s behavior or something else like spawning a new program? If so I am not sure if it can be done in a better way.

**Comment by xaizek** [comment]
`dirvish` operates within `vim`. You can see the terminal here, because that's how `vim` works by hiding its interface and then running another process on `:!cmd`.

`:terminal` isn't actually used by default in non-graphical `vim`, to force it set
```vim
let g:vifm_embed_term = 1
```
in your `vimrc` file. With this previous state of the terminal shouldn't appear.

**Comment by varmanishant** [comment]
Thank you so much @xaizek.

**Comment by varmanishant** [comment]
If I add:

`call feedkeys("<esc>")` after https://github.com/vifm/vifm.vim/blob/master/plugin/vifm.vim#L94 it works. But that feels hacky.

**Comment by xaizek** [comment]
I see the same weird behaviour. I don't really know what might be causing it. It looks as if Vim needs to visit event loop to realize that window isn't a terminal anymore and restore bindings, but I'm not sure how to check for this if the state is actually restored the moment a key is pressed.

**Comment by varmanishant** [comment]
Oh, so probably just send a PR with the hack?

**Comment by xaizek** [comment]
`call feedkeys("\<esc>", 'n')` is better. You can send a pull request, but I'll need to think if this should be addressed by the plugin or just in Vim.

**Comment by varmanishant** [comment]
Okay here is the [PR](https://github.com/vifm/vifm.vim/pull/21). Let me know if you find the real problem!. If it is a `vim` issue then we need something similar after closing the terminal and reported to vim itself.

**Comment by varmanishant** [comment]
@xaizek , shall I close the PR (not the issue) as this is only a temporary workaround? Will wait for a proper fix hopefully in Vim.

**Comment by xaizek** [comment]
I didn't get to debug it and if it's a Vim's bug, it at least needs to be reported before waiting for it to be fixed.

**Comment by xaizek** [comment]
Can you still reproduce it? It looks like I can't. It doesn't seem to be caused by the plugin, because I can't reproduce it on version from April. However, I did upgrade Vim since than and now using 8.1.1583. If you can reproduce it, what's your version of Vim?

**Comment by varmanishant** [comment]
I am able to reproduce this in Vim version 8.1.1694. It is not caused by the plugin because it works in Neovim. I could remove the PR; I guess it could be slight problem in Vim's terminal emulator?

**Comment by xaizek** [comment]
Thanks for testing. `let g:vifm_embed_split = 1` in my configuration prevented it from happening, so it's still there. I'll try to look into it a bit.

**Comment by xaizek** [comment]
I found vim/vim#3522 bug report, which is what we're seeing here and sent vim/vim#4683 which tries to fix it.

**Comment by varmanishant** [comment]
Thank you @xaizek . Appreciate! I will go through these PR's and try to understand what is happening! I'll remove the MR -- the details are here anyway if anyone wants this quick hack.

**Comment by xaizek** [comment]
The change got applied in [8.1.1703](https://github.com/vim/vim/releases/tag/v8.1.1703). Does it work for you?

**Comment by varmanishant** [comment]
Yes, it works fine in [8.1.1703](https://github.com/vim/vim/releases/tag/v8.1.1703). What does `ins_char_typebuf(K_IGNORE);` do? I am just curious. I think I can close this PR!

**Comment by xaizek** [comment]
It inserts a fake character into buffer used for reading input. As I understand there is a separate loop for reading input in a terminal mode, the problem is that it does an extra iteration before exiting and this iteration consumes (somehow conditionally, as builtin mappings aren't affected) and throws away one character sometimes. Putting a fake character that doesn't mean anything fixes this.
